custom/enikshay/integrations/nikshay/repeaters.py

```diff
@@ -128,6 +134,49 @@ def allowed_to_forward(self, episode_case):
         )
 
 
+class NikshayFollowupRepeater(CaseRepeater):
+    followup_for_tests = ['end_of_ip', 'end_of_cp']
+
+    class Meta(object):
+        app_label = 'repeaters'
+
+    include_app_id_param = False
+    friendly_name = _("Forward eNikshay Patient's Follow Ups to Nikshay")
+
+    @classmethod
+    def available_for_domain(cls, domain):
+        return NIKSHAY_INTEGRATION.enabled(domain)
+
+    @classmethod
+    def get_custom_url(cls, domain):
+        from custom.enikshay.integrations.nikshay.views import NikshayPatientFollowupRepeaterView
+        return reverse(NikshayPatientFollowupRepeaterView.urlname, args=[domain])
+
+    def allowed_to_forward(self, test_case):
+        # test.date_tested populates and test.nikshay_registered is false
+        # test.test_type_value = microscopy-zn or test.test_type_value = microscopy-fluorescent
+        # and episode.nikshay_registered is true
+        allowed_case_types_and_users = self._allowed_case_type(test_case) and self._allowed_user(test_case)
+        if allowed_case_types_and_users:
+            occurence_case = get_occurrence_case_from_test(test_case.domain, test_case.get_id)
+            episode_case = get_open_episode_case_from_occurrence(test_case.domain, occurence_case.get_id)
+            test_case_properties = test_case.dynamic_case_properties()
+            episode_case_properties = episode_case.dynamic_case_properties()
+            return (
+                date_tested_added_for_test(test_case) and
```

Comment:
can you use the `case_properties_changed` method instead of this? 
Also, put this after the other conditions, because I think it does a bunch of db lookups.